src/route.lua

```diff
@@ -17,7 +18,11 @@ function Route.new(route)
 
   for _, path in ipairs(route.paths) do
     if str_byte(path) ~= BYTE_SLASH then
-      return nil, "path must start with /"
+      return nil, "path must starts with /"
+    end
+    local _, pattern_idx = str_find(path, "{%*[^}]*}", 0, false)
```

Comment:
It seems to still work, but I think the `0` should be a `1` here, since it's the index of the character to start at, and lua strings indexes start at 1. 1 and false are the default, though, so this could just be `str_find(path, "{%*[^}]*}")`